docs/docs/reference/type-lambdas-spec.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: doc-page
+title: "Type Lambdas - More Details"
+---
+
+## Syntax
+
+```
+Type              ::=  ... |  HkTypeParamClause ‘=>’ Type
+HkTypeParamClause ::=  ‘[’ HkTypeParam {‘,’ HkTypeParam} ‘]’
+HkTypeParam       ::=  {Annotation} [‘+’ | ‘-’] (Id[HkTypeParamClause] | ‘_’) TypeBounds
+TypeBounds        ::=  [‘>:’ Type] [‘<:’ Type]
+```
+
+### Type Checking
+
+A type lambda such `[X] => F[X]` defines a function from types to types. The parameter(s) may carry bounds and variance annotations.
+If a parameter is is bounded, as in `[X >: L <: H] => F[X]` it is checked that arguments to the parameters conform to the bounds `L` and `H`.
+Only the upper bound `H` can be F-bounded, i.e. `X` can appear in it.
+
+A variance annotation on a parameter indicates a subtyping relationship on type instances. For instance, given
+```scala
+type TL1 = [+A] => F[A]
+type TL2 = [-A] => F[A]
+```
+and two types `S <: T`, we have
+```scala
+TL1[S] <: TL1[T]
+TL2[T] <: TL2[S]
+```
+It is checked that variance annotations on parameters of type lambdas are respected by the parameter occurrences on the type lambda's body.
+
+**Note** No requirements hold for the variances of occurrences of type variables in their bounds. It is an open question whether we need to impose additional requirements here
+(`scalac` doesn't check variances in bounds either).
+
+## Subtyping Rules
+
+Assume two type lambdas
+```scala
+type TL1  =  [v1 X >: L1 <: U1] => R1
+type TL2  =  [v2 X >: L2 <: U2] => R2
+```
+where `v1` and `v2` are optional variance annotations: `+`, `-`, or absent.
+Then `TL1 <: TL2`, if the type interval `L2..U2` is contained in the type interval `L1..U1` (i.e.
+`L1 <: L2` and `U2 <: U1`), and either `v2` is absent or `v1 = v2`.
```

Comment:
The subtyping rule says nothing about `R1` and `R2`. That looks suspicious to me. Shouldn't `R1 <: R2`?